controllers/syncconfig_status.go

```diff
@@ -0,0 +1,84 @@
+package controllers
+
+import (
+	"github.com/vshn/espejo/api/v1alpha1"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (r *SyncConfigReconciler) updateStatus(rc *ReconciliationContext) error {
+	newStatus := v1alpha1.SyncConfigStatus{
+		SynchronizedItemCount: rc.syncCount,
+		DeletedItemCount:      rc.deleteCount,
+		FailedItemCount:       rc.failCount,
+	}
+	for _, condition := range rc.conditions {
+		newStatus.Conditions = append(newStatus.Conditions, condition)
+	}
+
+	rc.cfg.Status = newStatus
+	err := r.Client.Status().Update(rc.ctx, rc.cfg)
+	if err != nil {
+		r.Log.Error(err, "Could not update SyncConfig.", getLoggingKeysAndValuesForSyncConfig(rc.cfg)...)
+		return err
+	}
+	r.Log.WithValues("syncCount", rc.syncCount, "deleteCount", rc.deleteCount, "failCount", rc.failCount).
+		Info("Updated SyncConfig status.", getLoggingKeysAndValuesForSyncConfig(rc.cfg)...)
+	return nil
+}
+
+// SetStatusCondition adds the given condition to the status condition of the SyncConfig. Overwrites existing conditions
+// of the same type.
+func (rc *ReconciliationContext) SetStatusCondition(condition v1alpha1.SyncConfigCondition) {
```

Comment:
We can use this helper to handle this: https://github.com/kubernetes/kubernetes/pull/92717/files#diff-92b43b13f51cc268adcd402276d8cf5e8076f0a32705820a493a40b7fa0e918cR30